src/sympc/protocol/aby3/aby3.py

```diff
@@ -237,24 +239,39 @@ def local_decomposition(
         rank = session.rank
         zero = torch.zeros(x.shares[0].shape).type(tensor_type)
 
-        shares = [[zero.clone(), zero.clone()] for i in range(NR_PARTIES)]
+        # Similar to triples, we have instances for the shares generated.
+        share_lst: List[List[List[ReplicatedSharedTensor]]] = []
+
+        input_rst = []
+        if bitwise:
+            ring_bits = get_nr_bits(session.ring_size)  # for bit-wise decomposition
+            for i in range(ring_bits):
+                input_rst.append(x.bit_extraction(i))
+        else:
+            input_rst.append(x)
+
+        for share in input_rst:
+            shares = [[zero.clone(), zero.clone()] for i in range(NR_PARTIES)]
```

Comment:
Q: Can't we use the PRZS for generating the zeros?

Reply:
Good question George,
Yeah we could definitely do `PRZS`.
But, I think it won't give a gain in `computational` and `security` standpoint.

When parties split shares of (x1,x2,x3)
x1 shares = (x1,0,0)
x2 shares= (0,x2,0)
x3 shares = (0,0,x3)


1.)`Security`  ; from a security perspective , each party knows other parties share places(meaning the places where they have `one` and actual share value),we do not use it to mask any secret value,the zero we can assume it as an implicit value.

2.)`Computational` : I think even generating przs is bit compute intensive,compared to generating zero directly by torch.

Reply:
Yep. As we discussed on Slack. I was thinking that maybe someone outside the protocol might watch, but the assumption in MPC scenarios is that the communication channel does not have any man in the middle